lib/rabbit_feed/producer_connection.rb

```diff
@@ -13,53 +12,37 @@ class ProducerConnection
       no_declare:  false,  # Create the exchange if it does not exist
     }.freeze
 
-    attr_reader :exchange
-
     def self.handle_returned_message return_info, content
       RabbitFeed.log.error "Handling returned message on #{self.to_s} details: #{return_info}..."
       RabbitFeed.exception_notify (ReturnedMessageError.new return_info)
     end
 
-    def initialize channel
-      RabbitFeed.log.debug "Declaring exchange on #{self.to_s} (channel #{channel.id}) named: #{RabbitFeed.configuration.exchange} with options: #{exchange_options}..."
+    def initialize
+      super
       @exchange = channel.exchange RabbitFeed.configuration.exchange, exchange_options
-
       exchange.on_return do |return_info, properties, content|
         RabbitFeed::ProducerConnection.handle_returned_message return_info, content
       end
     end
 
-    def self.publish message, options
-      retry_on_closed_connection do
-        with_connection do |producer_connection|
-          retry_on_exception do
-            producer_connection.publish message, options
-          end
-        end
-      end
-    end
-
     def publish message, options
-      # It's critical to dup the options for the sake of retries, as bunny modifies this hash
-      bunny_options = (options.merge PUBLISH_OPTIONS)
+      thread_safe do
```

Comment:
Looks like `publish` is the only public method on this object, so in the current implementation it's thread safe.